corehq/apps/app_manager/static/app_manager/js/forms/case_config_ui.js

```diff
@@ -385,6 +385,25 @@ hqDefine('app_manager/js/forms/case_config_ui', function () {
                 saveButton.fire('change');
             };
 
+            self.switchSaveOnlyIfEdited = function (property, event) {
+                if (!self.hasPrivilege) return;
+                var checked = event.target.checked;
+                if (checked) {
+                    hqImport('analytix/js/google').track.event('Case Management', analyticsAction, 'Checked "Save only if edited"');
+                } else {
+                    hqImport('analytix/js/google').track.event('Case Management', analyticsAction, 'Unchecked "Save only if edited"');
+                }
+                var updatedCaseProp = caseProperty.wrap({
```

Comment:
I'm a bit confused by the wrap and replacement. It looks like `save_only_if_edited` is an observable (it's in the `mapping` for `casePropertyBase`), so the js data should be updated automatically, and you should be able to subscribe to `save_only_if_edited` it in order to fire code when it changes, instead of setting up a click handler. The subscribe callback would just need to track analytics and maybe fire the save button change. Thoughts?

Reply:
I think that would be a smoother, cleaner approach. My confusion with the mapping code and limited knowledge of knockout subscribing is showing itself. I will change!

Reply:
@orangejenny So I tried to implement the above and noticed some barriers...

Each property in `visible_case_properties` serves as the context inside the HTML template. I could subscribe to this property's `save_only_if_edited` field and trigger a function. But I realized there is also the regular `case_properties` (the _real_ value tracker) mapping which needs to be updated. And looking at the ways other property values in this part of the code are tracked, I noticed  `removeProperty()` and `switchSaveOnlyIfEdited()` manually update both property arrays. So considering that I'd need to update a second array with the creation of a `caseProperty` via the wrap and replacement anyways and that what I have now is following the organization of the rest of the code, I figure it's better to follow convention and keep it the way it is, unless you have another idea.

Reply:
I see what you mean...looking at `removeProperty` and `addProperty`, I'm persuaded.

One smaller modification to consider is passing something like `_.extend(ko.mapping.toJS(property), {save_only_if_edited: checked})` to `wrap` instead of getting each property manually, so this doesn't need to be updated the next time someone adds a new attribute to `caseProperty`.

Reply:
The `extend` method also just looked way cleaner, so I added and it worked great immediately, thanks!